src/components/TextField/TextField.tsx

```diff
@@ -32,7 +49,44 @@ const replaceTargetValueWithUnformattedValue = ({
   values,
   sourceInfo,
 }: ReplaceTargetValueWithUnformattedValueProps) => {
-  sourceInfo.event.target.value = values.value;
+  (sourceInfo.event as React.ChangeEvent<HTMLInputElement>).target.value =
+    values.value.trim();
+};
+
+// Mitigates this issue: https://github.com/s-yadav/react-number-format/issues/694
+// Remove if fixed
+const numericKeyDown = (
```

Comment:
Could we contribute this fix upstream? :thinking:

Reply:
Yes, but it would have to be implemented slightly differently

Reply:
That serves more than just us, so I think we should go ahead with trying to fix this upstream if possible!